airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -118,11 +123,51 @@ def __init__(
         self.dynamodb_scan_kwargs = dynamodb_scan_kwargs
         self.s3_bucket_name = s3_bucket_name
         self.s3_key_prefix = s3_key_prefix
+        self.export_time = export_time
+        self.export_format = export_format
 
     def execute(self, context: Context) -> None:
         hook = DynamoDBHook(aws_conn_id=self.source_aws_conn_id)
-        table = hook.get_conn().Table(self.dynamodb_table_name)
+        if self.export_time:
+            self._export_table_to_point_in_time(hook=hook)
+        else:
+            self._export_entire_data(hook=hook)
+
+    def _export_table_to_point_in_time(self, hook: DynamoDBHook):
```

Comment:
I would change the logic of this function. I would wait for the export to be done only if specified by the user with a new flag `wait_for_completion`. If `False`, then just trigger the export without waiting. If `True`, wait for the export but use a waiter in order to do so. Do not use a `while True`. See [here](https://github.com/apache/airflow/tree/main/airflow/providers/amazon/aws/waiters) how to create a custom waiter

Reply:
Thanks for the suggestions. :)
1. Updated code to implement `waiter` logic. This is much better now!
2. I feel if we add `wait_for_completion` it will be only applicable if a user is specifying `export_time` that is using the `ExportTableToPointInTime` API via a waiter. The older path cannot accommodate this logic since we are getting data to the worker node and pushing in a loop. That would be a little inconsistent IMO. WDYT?